src/NuGet.Core/NuGet.Protocol.Core.v3/HttpSource/HttpSource.cs

```diff
@@ -162,6 +162,12 @@ public HttpSource(PackageSource source, Func<Task<HttpHandlerResource>> messageH
                             return new HttpSourceResult(HttpSourceResultStatus.NotFound);
                         }
 
+                        if (response.StatusCode == HttpStatusCode.NoContent)
```

Comment:
No, callers just need a way to know that it was a 204 and not a 200 with an empty page.